src/app/page.tsx

```diff
@@ -1,116 +1,121 @@
+    ]);
+    const priceInWei = claimCondition?.pricePerToken;
+    const currency = claimCondition?.currency;
+    const currencyMetadata = currency
+        ? await getCurrencyMetadata({ contract: getContract({ address: currency, chain, client }) })
+        : undefined;
+    return {
+        displayName: metadata?.data?.name,
+        description: metadata?.data?.description,
+        pricePerToken: currencyMetadata && priceInWei ? Number(toTokens(priceInWei, currencyMetadata.decimals)) : null,
+        currencyMetadata,
+        metadata,
+        claimCondition,
+    };
+}
 
-	const currencyContract = getContract({
-		address: currency || "",
-		chain,
-		client,
-	});
+async function getERC1155Info() {
+    const [nft, claimCondition] = await Promise.all([
+        getNFT({ contract, tokenId }),
+        getActiveClaimCondition1155({ contract, tokenId }),
+    ]);
+    const priceInWei = claimCondition?.pricePerToken;
+    const currency = claimCondition?.currency;
+    const currencyMetadata = currency
+        ? await getCurrencyMetadata({ contract: getContract({ address: currency, chain, client }) })
+        : undefined;
+    return {
+        displayName: nft?.metadata.name,
+        description: nft?.metadata.description,
+        pricePerToken: currencyMetadata && priceInWei ? Number(toTokens(priceInWei, currencyMetadata.decimals)) : null,
+        currencyMetadata,
+        nft,
+        claimCondition,
+    };
+}
 
-	const currencyMetadata = useReadContract(getCurrencyMetadata, {
-		contract: currencyContract,
-		queryOptions: { enabled: !!currency },
-	});
+type ERCInfo = {
+    displayName?: string;
+    description?: string;
+    pricePerToken?: number | null;
+    currencyMetadata?: any;
+    metadata?: any;
+    claimCondition?: any;
+    nft?: any;
+};
 
-	const currencySymbol = currencyMetadata.data?.symbol || "";
+export default async function Home() {
+    const ercType = await getERCType();
+    let info: ERCInfo = {};
 
-	const pricePerToken =
-		currencyMetadata.data && priceInWei !== null && priceInWei !== undefined
-			? Number(toTokens(priceInWei, currencyMetadata.data.decimals))
-			: null;
+    switch (ercType) {
+        case "ERC20":
+            info = await getERC20Info();
+            console.log("ERC20", info);
+            break;
+        case "ERC721":
+            info = await getERC721Info();
+            console.log("ERC721", info
+            );
+            break;
+        case "ERC1155":
+            info = await getERC1155Info();
+            console.log("ERC1155", info);
+            break;
+    }
 
-	return (
-		<NftMint
-			contract={contract}
-			displayName={displayName || ""}
-			contractImage={contractMetadataQuery.data?.image || ""}
-			description={description || ""}
-			currencySymbol={currencySymbol}
-			pricePerToken={pricePerToken}
-			isERC1155={!!isERC1155Query.data}
-			isERC721={!!isERC721Query.data}
-			tokenId={tokenId}
-		/>
-	);
+    return (
+        <NftMint
```

Comment:
feels like the component itself could use a bit of cleanup in the same way?

like one component per ERC, with common components inside?